src/events/IonEventStream.ts

```diff
@@ -229,39 +230,40 @@ export class IonEventStream {
                             break;
                         }
                         default: {
-                            this.eventStream.push(this.eventFactory.makeEvent(IonEventType.SCALAR, tid, this.reader.fieldName(), this.reader.depth(), this.reader.annotations(), false, this.reader.value()));
+                            this.events.push(this.eventFactory.makeEvent(IonEventType.SCALAR, tid, this.reader.fieldName(), this.reader.depth(), this.reader.annotations(), false, this.reader.value()));
                             break;
                         }
                     }
                 }
                 tid = this.reader.next();
             }
-        } catch (er) {
-            throw new Error("READ " + er.message);
+        } catch (error) {
+            // This Error will be used by the test-driver to differentiate errors using error types(prefix).
+            throw new EventStreamError("READ" , error.message , this.events.length);
```

Comment:
This is a good change -- I like that `READ` is an independently addressable/testable part of the `Error` now.

Instead of the string `"READ"` or `"WRITE"`, could this just be an instance of `ErrorType`?

Reply:
You mean creating two different Error classes for READ and WRITE?

Reply:
Oh I understand will do that.

Reply:
> I could declare Error Type in EventStreamError if we want to.

This sounds okay to me. This is a pretty minor concern anyway, we could merge the change even if we don't want to move `ErrorType`.